app/cdap/components/Connections/Browser/GenericBrowser/apiHelpers.ts

```diff
@@ -35,6 +35,10 @@ export function exploreConnection({ connectionid, path = '/' }) {
   ).toPromise();
 }
 
+export function getApiErrorMessage(e) {
+  return e?.message || e?.response?.message || e?.statusCode || 'Unknown error';
```

Comment:
should this now check if `e` is a string and if so, return it as is, to ensure `getApiErrorMessage(getApiErrorMessage(e)) == getApiErrorMessage(e)`? this may help the other comment